app/models/host/discovered.rb

```diff
@@ -13,8 +12,8 @@ class Host::Discovered < ::Host::Base
 
   scoped_search :on => :name, :complete_value => true, :default_order => true
   scoped_search :on => :last_report, :complete_value => true
-  scoped_search :on => :ip, :complete_value => true
-  scoped_search :on => :mac, :complete_value => true
+  scoped_search :in => :primary_interface, :on => :ip, :complete_value => true
+  scoped_search :in => :primary_interface, :on => :mac, :complete_value => true
```

Comment:
don't we need to add back the subnet for search?
also, should we search in => :interfaces instead of primary?